### PR TITLE
[scip] add v9.2.0

### PR DESCRIPTION
**Pull request by hedtke**
### Summary
Changes to recipe:  **scip/9.2.0**

#### Motivation
SCIP v9.2.0 was released

#### Details
I added scip 9.2.0. SCIP changed one of its default settings, so I moved that to the conandata file. I added a patch as the new upstream CMakeLists contains a wrong path, upstream issue https://github.com/scipopt/scip/issues/117 created

This would make #24682 obsolete

The CI will fail until soplex/7.1.2 is released, see #25871 

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/scip/all/conandata.yml [hunk 1]
```diff
@@ -11,3 +14,24 @@ sources:
...
     url: "https://github.com/scipopt/scip/archive/refs/tags/v803.tar.gz"
     sha256: "fe7636f8165a8c9298ff55ed3220d084d4ea31ba9b69d2733beec53e0e4335d6"
+patches:
+  "9.2.0":
+    - patch_file: "patches/0001-bliss-include-dir.patch"
+      patch_description: "Change hard-coded paths to conan includes"
+      patch_type: "conan"
+version_mappings:
+  "9.2.0":
+    soplex: "7.1.2"
```
Comment (jcar87) [inline recipes/scip/all/conandata.yml:24]: mmm this version does not seem to exist

```
  soplex
    soplex/6.0.3
    soplex/6.0.4
    soplex/7.0.0
    soplex/7.0.1
    soplex/7.1.0
    soplex/7.1.1
```

File: recipes/scip/all/conanfile.py [hunk 2]
```diff
@@ -84,13 +80,21 @@ def config_options(self):
...
+
         if self.options.with_gmp:
             self.requires("gmp/6.3.0")
         if self.options.with_sym == "bliss":
             self.requires("bliss/0.77")
-        self.requires(f"soplex/{self.soplex_version_belonging_to_me[self.version]}")
+        _mapping_requires("soplex")
         self.requires("zlib/[>=1.2.11 <2]")
 
+    def config_options(self):
```
Comment (jcar87) [inline recipes/scip/all/conanfile.py:94]: this method is being redefined from elsewhere

File: recipes/scip/all/conandata.yml (hunk 1 above)
Comment (hedtke) [reply recipes/scip/all/conandata.yml:24]: This MR depends on the one adding scip/7.1.2

File: recipes/scip/all/conanfile.py (hunk 2 above)
Comment (hedtke) [reply recipes/scip/all/conanfile.py:94]: I don't understand this comment. What am I doing wrong?

Comment (jcar87) [reply recipes/scip/all/conanfile.py:94]: Sorry - it means that the recipe contains to definitions of the `config_options(self)` method - so one is overriding the other

Comment (hedtke) [reply recipes/scip/all/conanfile.py:90]: Whupps, I added a commit to merge both

Comment (jcar87) [reply recipes/scip/all/conanfile.py:90]: Thanks! :)

File: recipes/scip/all/conandata.yml (hunk 1 above)
Comment (jcar87) [reply recipes/scip/all/conandata.yml:24]: Excellent - relaunching this so that it gets built now

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Review by jcar87 — APPROVED** [review]

**Pull request merged**